### PR TITLE
[SPDBT-3077] cancel and inactive spd_app when delete non-swl bizContact

### PR DESCRIPTION
**Pull request by esdd1995**
getting `FirstOrDefault()` from the `spd_businesscontact_spd_application` collection, then cancel it.

### CODE REVIEW

File: src/Spd.Resource.Repository/BizContact/BizContactRepository.cs [hunk 1]
```diff
@@ -122,8 +122,22 @@ public async Task<IEnumerable<BizContactResp>> QueryBizContactsAsync(BizContactQ
         private async Task<Guid?> DeleteBizContactAsync(BizContactDeleteCmd cmd, CancellationToken ct)
         {
             spd_businesscontact? bizContact = await _context.GetBizContactById(cmd.BizContactId, ct);
+
+            if (bizContact == null)
+                throw new ApiException(HttpStatusCode.BadRequest, $"business contact with id {cmd.BizContactId} not found.");
+
+            spd_application? app = _context.spd_applications
+                .Expand(a => a.spd_businesscontact_spd_application)
+                .Where(x => x.spd_businesscontact_spd_application.Any(y=>y.spd_businesscontactid == cmd.BizContactId)).FirstOrDefault();
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/BizContact/BizContactRepository.cs:131]: I think the better way is to check if the application is in Incomplete, draft, waiting for payment, or Verfiying Identity, then set the applications to cancelled status.

**Review by peggy-quartech — CHANGES_REQUESTED** [review]
Please change. Sorry, it is different than I told you originally.

File: src/Spd.Resource.Repository/BizContact/BizContactRepository.cs (hunk 1 above)
Comment (esdd1995) [reply src/Spd.Resource.Repository/BizContact/BizContactRepository.cs:131]: and if app was not in these status, should we throw exception and prevent deactivating bizContact and app, or just deactivating bizContact, and do nothing with the app?

Comment (peggy-quartech) [reply src/Spd.Resource.Repository/BizContact/BizContactRepository.cs:131]: just deactivating bizContact

Comment (peggy-quartech) [reply src/Spd.Resource.Repository/BizContact/BizContactRepository.cs:131]: This is a very rare scenario, it is for business manager send out invitation, and cm almost complete submit app, then business manager remove him. There is no specific requirement about what should we do for this scenario, so, I do not want to over-engineering it. Let's make it simple for now.

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**